src/site/antora/modules/ROOT/pages/manual/flowtracing.adoc

```diff
@@ -14,109 +14,94 @@
-the visibility of application processes. With this technique, developers can track 
-data flow through their application using unique methods that log entry 
-and exit points within the code.
+xref:manual/api.adoc[Log4j API] provides convenience logging methods to aid the tracking of the data flow through an application, which is referred to as *flow tracing*.
+<<impl,Logging implementations>> can choose to <<log-events,generate specialized log events>> allowing users to handle these messages different from the rest – see <<example>>.
 
-These methods are:
+Flow tracing is known to help with the following use cases:
 
-* `traceEntry()` (Deprecated `entry()`)
-* `traceExit()` (Deprecated: `exit()`)
-* `throwing()`
-* `catching()`
+* Troubleshooting without requiring a debugging session
+* Helping educate new developers in learning the application
 
-With these methods, we can investigate environments where traditional debugging is not possible, 
-such as in production or during live application monitoring. 
-Furthermore, new developers can be educated on the application's behavior by examining the logs.
+[#usage]
+== Usage
 
-Flow tracing offers a structured approach to all this.
+link:../javadoc/log4j-api/org/apache/logging/log4j/Logger.html[`Logger`] provides following methods for flow tracing purposes:
 
-== Flow Tracing Methods
+`traceEntry()`:: Marks the entry to a block
+`traceExit()`:: Marks the exit from a block
+`catching()`:: Reports caught exceptions
+`throwing()`:: Reports exceptions that are either discarded or unlikely to be handled
 
-The methods often used are `traceEntry()` and `traceExit()`.
-As the name suggests, the "entry" method is used at the beginning and the "exit" method at the end of a method.
+The most used `traceEntry()` and `traceExit()` methods are intended to mark the _entry_ and _exit_ points of a particular block of code:
 
+[#example-trace]
+.Example `traceEntry()` and `traceExit()` usage
 [source, java]
 ----
 public void someMethod() {
-    logger.traceEntry(); <1>
+    LOGGER.traceEntry(); //<1>
     // method body
-    logger.traceExit(); <2>
+    LOGGER.traceExit(); //<2>
 }
 ----
-<1> The `entry()` method is called at the beginning of the method.
-<2> The `exit()` method is called at the end of the method.
+<1> `traceEntry()` marks the entry to the block
+<2> `traceExit()` marks the exit from the block
 
-Developers can call both `traceEntry()` and `traceExit()` methods with or without parameters.
-In the case of `traceEntry()`, it makes sense to pass the method parameters on as arguments.
+Both `traceEntry()` and `traceExit()` also accept parameters.
+You can use them to track the input and output of the associated block:
 
+[#example-trace-args]
+.Example `traceEntry()` and `traceExit()` usage with arguments
 [source, java]
 ----
-public void someMethod(String param) {
-    logger.traceEntry(param); <1>
+public String someMethod(String input) {
+    logger.traceEntry(input); // <1>
```

Comment:
```suggestion
    logger.traceEntry(null, input); // <1>
```

This is a call to [`traceEntry(String format, Object... params)`](https://logging.apache.org/log4j/2.x/javadoc/log4j-api/org/apache/logging/log4j/Logger.html#traceEntry(java.lang.String,java.lang.Object...), so passing user-generated data as the format string is dangerous. :wink:

If the format is `null` the default `FlowMessageFactory` generates the appropriate format. In this case `params({})`.